runtime/compiler/optimizer/JProfilingValue.cpp

```diff
@@ -434,13 +461,13 @@ TR_JProfilingValue::lowerCalls()
  */
 bool
 TR_JProfilingValue::addProfilingTrees(
```

Comment:
Given the complexity of the transform could you add some ascii art to show the before and after block structures with the relevant semantic descriptions of the blocks? could you also make sure the labels in the diagram match the local vars below so it is easy to find the different blocks?

Reply:
I have modified the block layout to reflect trees which are closer to what it would look like in actual log. Also added variable names used for different blocks and nodes.